plugin.video.esa/resources/lib/lutil.py

```diff
@@ -0,0 +1,210 @@
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+   Description:
+   These funtions are called from the main plugin module, aimed to ease and simplify the plugin development process.
+   Release 1.3.0
+'''
+
+# First of all We must import all the libraries used for plugin development.
+import sys, re, os
+from urllib.parse import urlencode, quote_plus, unquote_plus
+from urllib.request import urlopen, Request
+import xbmcplugin, xbmcaddon, xbmcgui, xbmcaddon, xbmc
+
+debug_enable = False # The debug logs are disabled by default.
+fanart_file = "" # Initialize the global var for the fanart file location.
+
+
+# This function returns the plugin handle to main module.
+def get_plugin_handle():
+    return int(sys.argv[1])
+
+
+# This function returns the plugin settings object to main module.
+def get_plugin_settings(plugin_id=""):
+    return xbmcaddon.Addon(id=plugin_id)
+
+
+# This function returns the current system language.
+def get_system_language():
+    return xbmc.getLanguage()
+
+
+# This function sets the debug_enable var to log everything if debug option is true.
+def set_debug_mode(debug_flag=""):
+    global debug_enable
+    if debug_flag == "true":
+        debug_enable = True
+
+
+# This function setup the file and global plugin fanart.
+def set_fanart_file(root_path=""):
+    global fanart_file
+    fanart_file = os.path.join(root_path, "fanart.jpg")
+    xbmcplugin.setPluginFanart(int(sys.argv[1]), fanart_file)
+
+
+# This function logs the messages into the main KODI log file. Called from main plugin module.
+def log(message):
+    if debug_enable:
+        xbmc.log("%s" % message, level=xbmc.LOGINFO)
+
+
+# This function logs the messages into the main KODI log file. Called from the libraries module by other functions.
+def _log(message):
+    if debug_enable:
+        xbmc.log("lutils.%s" % message, level=xbmc.LOGINFO)
+
+
+# This function gets all the parameters passed to the plugin from KODI API and retuns a dictionary.
+# Example:
+# plugin://plugin.video.atactv/?parametro1=valor1&parametro2=valor2&parametro3
+def get_plugin_parms():
+    params = sys.argv[2]
+    _log("get_plugin_parms " + str(params))
+
+    pattern_params  = re.compile('[?&]([^=&]+)=?([^&]*)')
+    options = dict((parameter, unquote_plus(value)) for (parameter, value) in pattern_params.findall(params))
+
+    _log("get_plugin_parms " + repr(options))
+    return options
+
+
+# This function returns the URL decoded.
+def get_url_decoded(url):
+    _log('get_url_decoded URL: "%s"' % url)
+    return unquote_plus(url)
+
+
+# This function returns the URL encoded.
+def get_url_encoded(url):
+    _log('get_url_encoded URL: "%s"' % url)
+    return quote_plus(url)
+
+
+# This function sets the view mode into the video list.
+def set_view_mode(viewid):
```

Comment:
Not allowed please remove

Reply:
Sorry, I wasn't aware of that. As the function isn't used anyway I've removed it safely from the lutil.py file.